### PR TITLE
mpi/coll: Add support for comm_avg_ppn=any in the tuning json

### PR DESCRIPTION
**Pull request by rithwiktom**
## Pull Request Description

 `comm_avg_ppn=any` and `<` can be set in the json tuning file. Useful for selecting algo based on the ppn size.

## Author Checklist
* [ ] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [ ] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [ ] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [ ] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

File: src/mpi/coll/src/csel.c [hunk 1]
```diff
@@ -288,7 +299,9 @@ static void validate_tree(csel_node_s * node)
     if (node->type == CSEL_NODE_TYPE__OPERATOR__COMM_SIZE_ANY ||
         node->type == CSEL_NODE_TYPE__OPERATOR__AVG_MSG_SIZE_ANY ||
         node->type == CSEL_NODE_TYPE__OPERATOR__TOTAL_MSG_SIZE_ANY ||
-        node->type == CSEL_NODE_TYPE__OPERATOR__COUNT_ANY) {
+        node->type == CSEL_NODE_TYPE__OPERATOR__COUNT_ANY ||
+        node->type == CSEL_NODE_TYPE__OPERATOR__COMM_AVG_PPN_ANY) {
```
Comment (hzhou) [inline src/mpi/coll/src/csel.c:303]: The `_ANY` are always for the last "else"-branch, right? I wonder do we need all these tags or can we simply replace them with a `CSEL_NODE_TYPE__OPERATOR__ELSE`.

Just a comment. We can do this in a separate PR.

**Review by hzhou — APPROVED** [review]
LGTM. Approved pending tests

**Comment by hzhou** [comment]
test:mpich/ch3/most
test:mpich/ch4/most

File: src/mpi/coll/src/csel.c (hunk 1 above)
Comment (rithwiktom) [reply src/mpi/coll/src/csel.c:303]: I agree. That would be concise.
